src/runtime_src/core/pcie/driver/linux/xocl/subdev/hwmon_sdm.c

```diff
@@ -361,20 +361,28 @@ static int show_sensors_raw(struct xocl_hwmon_sdm *sdm, char *buf,
 {
 	xdev_handle_t xdev = xocl_get_xdev(sdm->pdev);
 	int ret = 0, kind;
-	size_t resp_len = RESP_LEN;
+	size_t resp_len = 0;
 	char* sdr_buf;
 	int repo_type;
 	uint64_t data_args = 0;
 
 	mutex_lock(&sdm->sdm_lock);
+	/* Collect the SDR size from VMR */
+	xocl_xgq_collect_sensors_get_size(xdev, &resp_len, repo_id);
+
+	if (resp_len <= 0) {
```

Comment:
I would think there should be some "defensive" coding like
if ((resp_len <= 0) || (rep_len > RESP_LEN))
  goto fail